### PR TITLE
ELPP-3633 Run `vagrant up` using a container for the Python part

### PR DESCRIPTION
**Pull request by giorgiosironi**
`vagrant up` runs on the host, but delegates reading project data to `.project.py`, which we can execute inside a container. The image derived from `python:2.7` allows Mac users not to install Python for this use case.

## To test

1. Install `docker` if not already installed
2. Install `vagrant` (and `VirtualBox`) if not already installed (can't easily run in Docker, I think)
3. Run `vagrant up`
4. Choose `schematron-validator` which was our candidate example
5. See the whole provisioning succeed

### CODE REVIEW

**Comment by lsh-0** [comment]
Vagrant shouldn't depend on Docker, I would switch the executable inside Vagrant based on the presence of flag (`.use-docker.flag`).

**Comment by giorgiosironi** [comment]
There is also the `.use-python-3` flag, I'll add both.

**Comment by giorgiosironi** [comment]
Alfred doesn't have Docker available, at least in its master node.

File: Vagrantfile [hunk 1]
```diff
@@ -25,8 +25,14 @@ def runcmd(cmd)
 end
 
 def project_cmd(argstr)
-    cmd = "/bin/bash -c \"source venv/bin/activate && ./.project.py #{argstr}\""
-    #prn(cmd)
+    flag = ".use-docker.flag"
+    if File.exists?(flag)
```
Comment (lsh-0) [inline Vagrantfile]: :+1:

**Comment by giorgiosironi** [comment]
Note: the container is writing `__pycache__` folders and `.pyc` files in the source folder, mounted on it. Need to avoid that.

**Comment by lsh-0** [comment]
made some suggestions here:
https://github.com/elifesciences/builder/pull/348

**Comment by giorgiosironi** [comment]
Finally green.

**Pull request merged**